src/platform-aws.c

```diff
@@ -97,207 +102,168 @@ static const char* get_platform_type(void)
+			      sizeof(optval), optlen);
+		ret = -EINVAL;
+		goto exit;
+	} else if (optval) {
 		NCCL_OFI_WARN("FI_OPT_EFA_EMULATED_WRITE is true when the communication protocol is RDMA write.");
-		ret = ncclSystemError;
+		ret = -EINVAL;
 		goto exit;
 	}
 	NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET, "Get endpoint option FI_OPT_EFA_EMULATED_WRITE. optval: %d", 
 		       optval);
 #else
 	NCCL_OFI_WARN("FI_OPT_EFA_EMULATED_WRITE not declared when the communication protocol is RDMA write.");
-	ret = ncclSystemError;
+	ret = -EINVAL;
 	goto exit;
 #endif
+
 exit:
 	return ret;
 }
 
-static ncclResult_t configure_sendrecv_inorder(struct fid_ep *ep, bool is_init)
-{
-	int ret = ncclSuccess;
-#if HAVE_DECL_FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES
-	bool optval = true;
 
-	ret = fi_setopt(&ep->fid, FI_OPT_ENDPOINT,
-			FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES, 
-			&optval, sizeof(optval));
-	if (ret != 0 && ret != -FI_EOPNOTSUPP) {
-		NCCL_OFI_WARN("Couldn't set FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES. RC: %d, ERROR: %s",
-			      ret, fi_strerror(-ret));
-		ret = ncclSystemError;
-		goto exit;
-	}
+#if HAVE_CUDA
+static int configure_nccl_proto(bool have_ordering)
+{
+	int ret = 0;
 
-	/* If this is called during plugin initialization, set the global flag
-	 * sendrecv_support_ll128 to true if
-	 * FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES could be set to true,
-	 * otherwise keep it at its default value of false.
-	 */
-	if (is_init) {
-		if (ret == 0) {
-			sendrecv_support_ll128 = true;
+	if (!getenv("NCCL_PROTO")) {
+		NCCL_OFI_INFO(NCCL_INIT, "Setting NCCL_PROTO to \"simple\"");
+		ret = setenv("NCCL_PROTO", "simple", 0);
+		if (ret != 0) {
+			NCCL_OFI_WARN("Error setting NCCL_PROTO environment variable: %s",
+				      strerror(errno));
+			return -errno;
 		}
+	} else if (strcmp(getenv("NCCL_PROTO"), "simple") != 0) {
+		NCCL_OFI_WARN("NCCL_PROTO was set to \"LL/LL128\", but the Libfabric endpoint does not support 128 byte in-order aligned stores. This endpoint may corrupt data during communication");
 	}
-	/* If an endpoint supported SENDRECV LL128 during plugin initialization
-	 * but does not support it now, throw an error.
-	 */
-	else if (sendrecv_support_ll128 && ret == -FI_EOPNOTSUPP) {
-		NCCL_OFI_WARN("SENDRECV LL128 not supported while it was supported during initialization.");
-		ret = ncclSystemError;
-		goto exit;
-	}
-	/* If an endpoint did not support SENDRECV LL128 during plugin
-	 * initialization but supports it now, throw an error.
-	 */
-	else if (!sendrecv_support_ll128 && ret == 0) {
-		NCCL_OFI_WARN("SENDRECV LL128 supported while it not supported during initialization.");
-		ret = ncclSystemError;
-		goto exit;
-	}
-	NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET, "Set endpoint option FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES. optval: %d, RC: %d, ERROR: %s", 
-		       optval, ret, fi_strerror(-ret));
-	ret = ncclSuccess;
-exit:
-#endif
-	return ret;
+
+	return 0;
```

Comment:
nit: could replace `return 0` with `return ret`. Both will always return 0, just a matter of preference for style.

Reply:
I don't have strong opinions.  I usually return 0 in cases where I've left the `goto exit;` type pattern, so that if someone isn't careful about setting ret (there are times where you get an error, but can move on successfully), the function doesn't suddenly break.  But I'll change it if you have strong feelings.

Reply:
I'm fine with leaving it as is!